stat_posix.go

```diff
@@ -49,8 +49,8 @@ func isRegular(mode uint32) bool {
 	return mode&S_IFMT == syscall.S_IFREG
 }
 
-// toFileMode converts sftp filemode bits to the os.FileMode specification
-func toFileMode(mode uint32) os.FileMode {
+// ToFileMode converts sftp filemode bits to the os.FileMode specification
+func ToFileMode(mode uint32) os.FileMode {
```

Comment:
I need to look at the code again, but why is toFileMode and fromFileNode even in the build-tag dependent code?  The functionality described in the comment shouldn't be OS dependent.

Reply:
plan9 does not support: `S_ISUID`: `os.ModeSetuid`, `S_ISGID`:`os.ModeSetgid`, or `S_ISVTX`:`os.ModeSticky`

So, generic code cannot work.

Reply:
Hm, do the constants not exist on plan9?  I ask because I would expect the things that take a os.FileMode to deal with the flags not existing, instead of us needing to do this.

After all, this is converting to the os.FileMode specification, not the OS mode specification.

Reply:
Correct, the `S_` constants do not exist on `plan9`, so we were forced to implement them ourselves in this package for that OS.  As well, there is sort of an intent that `syscall` should not be used anymore anyways, as they’re moving onto a new organization of system internals.

But basically, there are a lot of _subtle_ issues here and bugs that we’ve had to squashed. I would really rather keep users isolated from these implementation details entirely. That way we can control very precisely how `os.FileMode` is handled and treated, so that we can fix bugs or redesigns (yes, we’ve had to do deep redesigns of this before) without worrying about compatibility promises.

The Go paradigm is “Simplicity is Complex”. The surface of an API should be simple, and all the complexity should be hidden away from the API user.